thijsvanloef/palworld-server-docker
# Cron update

## Context 

* adds a Cronjob to check for updates
* Cronjob can be set like the backup cronjob from @Dashboy1998
* Addressing #38  


## Choices

* using the steamapi to get the current build and compare it to the buildId in the file palworld\steamapps\appmanifest_2394010.acf. I am not sure it that is the best way to check. Maybe you have input on this.
* to "update" the server, the script will broadcast a message to the players and restart the server in x minutes. This will only work if the env UPDATE_ON_BOOT is set.
* I have set the default cron expression to once per hour and the warning minutes to 30 minutes

## Test instructions

1. Set  `UPDATE_CRON_EXPRESSION` to `* * * *`
2. Set `WARN_MINUTES` to `1`
3. check in the log that the cronjob is running correctly
4. (Optional) login to the server and create a character
5. change the buildId on line 11 in the file `palworld\steamapps\appmanifest_2394010.acf`` to any other value (for example 13289561)
6. Verify that the server restarts in one Minute after the cronjob has started

**A Real world test is still needed**

## Checklist before requesting a review
